tests/dragonfly/conftest.py

```diff
@@ -42,25 +44,28 @@ def pytest_configure(config):
         shutil.rmtree(BASE_LOG_DIR)
 
 
-# runs per test case
-def pytest_runtest_setup(item):
-    # Generate a unique directory name for each test based on its nodeid
+@pytest.fixture(scope="class")
+def df_log_dir(request):
```

Comment:
now we create a df_factory exactly once per test or TestClass, and df_log_dir is directly related to df_factory, so lets make it as a fixture dependency.